StoneMasonKarel.java

```diff
@@ -1,17 +1,70 @@
-/*
- * File: StoneMasonKarel.java
- * --------------------------
- * The StoneMasonKarel subclass as it appears here does nothing.
- * When you finish writing it, it should solve the "repair the quad"
- * problem from Assignment 1.  In addition to editing the program,
- * you should be sure to edit this comment so that it no longer
- * indicates that the program does nothing.
- */
-
-import stanford.karel.*;
-
-public class StoneMasonKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: StoneMasonKarel.java
+ * --------------------------
+ * The StoneMasonKarel subclass as it appears here does nothing.
+ * When you finish writing it, it should solve the "repair the quad"
+ * problem from Assignment 1.  In addition to editing the program,
+ * you should be sure to edit this comment so that it no longer
+ * indicates that the program does nothing.
+ */
+
+import stanford.karel.*;
+
+public class StoneMasonKarel extends SuperKarel {
+
+	// You fill in this part
+	public void run(){
+		for (int i=0; i<3; i++){
+		placeBeeperTillWall();
+		returnToStartingPoint();
+		moveToTheNextColumn();
+		
+		}
+		placeBeeperTillWall();
+			}
+
+	
+	
+
+	private void moveToTheNextColumn() {

```

Comment:
Great job on breaking down the problem and using function names that are easy to understand.